airflow/contrib/operators/gcs_list_operator.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class GoogleCloudStorageListOperator(BaseOperator):
+    """
+    List all objects from the bucket with the give string prefix and delimiter in name.
+
+    This operator returns a python list with the name of objects which can be used by
+     `xcom` in the downstream task.
```

Comment:
nit: it might be desirable to give an example on the list response pushed to XCOM (e.g., json encoded, etc) so others know how to consume the xcom value.

Reply:
I have the following example but I am unsure whether I should put this in docstrings or should I create a new file? Your suggestions are welcome.

```
# --------------------------------------------------------------------------------
# Functions
# --------------------------------------------------------------------------------


def get_values_from_xcom(**kwargs):
    ti = kwargs['ti']
    
    files = ti.xcom_pull(task_ids='GCS_List_Test_task')

    logging.info('Files in GCS: %s', files)
    
    for file in files:
        print(file)
        logging.info(file)
        
# --------------------------------------------------------------------------------
# Main DAG
# --------------------------------------------------------------------------------

with DAG('GCS_List_Test', default_args=default_args, schedule_interval=None) as dag:

    GCS_List_Test = GoogleCloudStorageListOperator(
        task_id='GCS_List_Test_task',
        bucket='test-bucket',
        prefix='sales_',
        delimiter='.csv',
        google_cloud_storage_conn_id='airflow-service-account'
    )

    pyth_op = PythonOperator(
        task_id='get_values',
        python_callable=get_values_from_xcom,
        provide_context=True
    )

    GCS_List_Test >> pyth_op
```